src/mbgl/annotation/annotation_tile.cpp

```diff
@@ -12,15 +12,13 @@ AnnotationTile::AnnotationTile(const OverscaledTileID& overscaledTileID,
                                TileObserver* observer_)
     : GeometryTile(overscaledTileID, AnnotationManager::SourceID, parameters, observer_),
       annotationManager(parameters.annotationManager) {
-    auto guard = annotationManager.lock();
-    if (annotationManager) {
+    if (auto guard = annotationManager.lock(); annotationManager) {
```

Comment:
clang-tidy will soon complain about assignments in if statements. Could you change it?